bench/spawn.go

```diff
@@ -239,3 +275,56 @@ func (cmd *SpawnCommand) spawnRemote(ctx context.Context) ([]BenchmarkResult, er
 
 	return results, nil
 }
+
+// getGithubContent using github api return url of a file to download and commit hash
+func (cmd *SpawnCommand) getGithubContent(ctx context.Context) (*GitContent, error) {
+	fileName := cmd.FileName
+	repo := cmd.Repo
+	path, err := url2.Parse(repo)
+	uri := path.Path
+	gitURL := fmt.Sprintf("https://api.github.com/repos%s/contents/%s", uri, fileName)
+
+	resp, err := http.Get(gitURL)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("getting file %v from %v, err: %v", fileName, repo, string(body))
+	}
+	var content GitContent
+	err = json.NewDecoder(resp.Body).Decode(&content)
+	if err != nil {
+		return nil, err
+	}
+	if content.DownloadURL == "" {
+		return nil, fmt.Errorf("No file to download")
+	}
+	return &content, nil
+}
+
+// downloadFile downloads git file and copy to temp file
+func (cmd *SpawnCommand) downloadFile(url string, fileName *os.File) error {
```

Comment:
instead of `filename *os.File` I think this function can just take a `writer io.Writer` - that will be more general, and easier to test.

Also I think it would be better if this was a standalone function rather than a method of SpawnCommand since it doesn't use anything from SpawnCommand